doc/bash-completion/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+IF(NOT LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_APPLE)
+
+	pkg_check_modules(BASH_COMPLETION bash-completion)
+	pkg_get_variable(BASHCOMP_PKG_PATH bash-completion completionsdir)
+	
+	GET_FILENAME_COMPONENT(BINARY_ADJ_TARGET "${CMAKE_BINARY_DIR}/../target" ABSOLUTE)
+	SET(BASHCOMP_WHITELIST_PREFIX "/usr;/usr/local;${BINARY_ADJ_TARGET}")
+	SET(BASHCOMP_WHITELIST_PREFIX_NOLOCAL "/usr;${BINARY_ADJ_TARGET}")
+	
+	SET(BASHCOMP_PKG_PATH "/etc/bash_completion.d")
+
+	IF("${BASHCOMP_PKG_PATH}" STREQUAL "/usr/share/bash-completion/completions")
+
+		# bash completion uses the "new" way of /usr/share/bash-completion
+		# this means both /usr/share/bash-completion and
+		# /usr/local/share/bash-completion are valid
+		# all other paths are user paths
+
+		# Add ../target to bash-completion whitelist for appimages
+		FOREACH(_prefix ${BASHCOMP_WHITELIST_PREFIX})
+		IF("${CMAKE_INSTALL_PREFIX}" MATCHES "^${_prefix}/?$")
+			SET(BASHCOMP_PATH_FINAL "${CMAKE_INSTALL_PREFIX}/share/bash-completion/completions")
+		ENDIF()
+		ENDFOREACH()
+		
+	ELSEIF("${BASHCOMP_PKG_PATH}" STREQUAL "/etc/bash_completion.d")
```

Comment:
Can you please explain the purpose of this logic?  If it's `/etc/bash_completion.d`, shouldn't we just blindly use it and white-list it?

Reply:
I wanted it to behave like the other logic, but don't install it if the install is going to `/usr/local` (imagine they're doing an architekture independent install to `/usr/local`, but an architecture depending one for the bash-completion to `/etc`). It's a contradiction to me, so I change the behavior for `/etc/bash_completion.d`.

Reply:
Sorry, I had made some bad logic suggestions on Discord... let's try this instead.  I think you'll find it covers all common scenarios and all edge-cases.

```cmake
# Detects the best place to install bash completion support
# - Windows is not supported
# - TODO: macOS support can be provided through Homebrew
IF(NOT LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_APPLE)

	# standard install location for most distributions
	SET(BASHCOMP_FALLBACK "/usr/share/bash-completion/completions")

	# check for common portable installation (CMAKE_INSTALL_PREFIX=../target)
	GET_FILENAME_COMPONENT(BINARY_ADJ_INSTALL "${CMAKE_BINARY_DIR}/../target" ABSOLUTE)
	IF("${BINARY_ADJ_INSTALL}" STREQUAL "${CMAKE_INSTALL_PREFIX}")
		SET(SKIP_PKG_CHECK TRUE)
	ENDIF()

	# check for other signs of user-level install
	SET(USER_LEVEL_INSTALL "/home/;/root/;/Users/")
	FOREACH(_startswith ${USER_LEVEL_INSTALL})
		IF("${CMAKE_INSTALL_PREFIX}" MATCHES  "^${_startswith}")
			SET(SKIP_PKG_CHECK TRUE)
			BREAK()
		ENDIF()
	ENDFOREACH()

	IF(SKIP_PKG_CHECK)
		SET(BASHCOMP_PKG_PATH "${CMAKE_INSTALL_PREFIX}${BASHCOMP_FALLBACK}")
	ELSE()
		PKG_CHECK_MODULES(BASH_COMPLETION bash-completion)
		PKG_GET_VARIABLE(BASHCOMP_PKG_PATH bash-completion completionsdir)
	ENDIF()

	# finally, fallback for older operating systems if using root-level install
	IF("${BASHCOMP_PKG_PATH}" STREQUAL "")
		SET(ROOT_LEVEL_INSTALL "/usr/;/usr/local/")
		FOREACH(_startswith ${ROOT_LEVEL_INSTALL})
			IF("${CMAKE_INSTALL_PREFIX}" MATCHES  "^${_startswith}")
				SET(BASHCOMP_PKG_PATH "${BASHCOMP_FALLBACK}")
				BREAK()
			ENDIF()
		ENDFOREACH()
	ENDIF()
	
	IF("${BASHCOMP_PKG_PATH}" STREQUAL "")
		MESSAGE("Unable to determine BASHCOMP_PKG_PATH.  Install manually from completions/lmms if needed.")
	ELSE()
		MESSAGE(STATUS "Bash completions will be installed to ${BASHCOMP_PKG_PATH}")
		INSTALL(FILES lmms DESTINATION "${BASHCOMP_PKG_PATH}" PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
	ENDIF()
ENDIF()
```

Reply:
@devnexen so that we can handle any edge-cases, can you tell us if this logic (at a glance) will work on OpenBSD?